private/buf/cmd/buf/command/breaking/breaking.go

```diff
@@ -251,9 +261,42 @@ func run(
 					imageWithConfig.ModuleOpaqueID(),
 				)
 			}
+			// Check commits exist for the module on the default label before we try to get the against image.
+			// TODO: remove this check once we have error details to indicate that the against input
+			// has no commits on the default label. We use the API directly here to avoid issues of conflating
+			// dependency errors to empty module errors.
+			commitServiceClient := moduleServiceProvider.V1CommitServiceClient(moduleFullName.Registry())
+			if _, err := commitServiceClient.GetCommits(
+				ctx,
+				connect.NewRequest(
+					&modulev1.GetCommitsRequest{
+						ResourceRefs: []*modulev1.ResourceRef{
+							{
+								Value: &modulev1.ResourceRef_Name_{
+									Name: &modulev1.ResourceRef_Name{
+										Owner:  moduleFullName.Owner(),
+										Module: moduleFullName.Name(),
+									},
+								},
+							},
+						},
+					},
+				),
+			); err != nil {
+				if connect.CodeOf(err) == connect.CodeFailedPrecondition {
```

Comment:
Why is an empty response returning failed precondition? Why not not found?

Also I would probably use list with a page size of 1, and see if the returned list has 0 elements.